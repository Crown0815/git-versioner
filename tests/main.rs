mod lib;

use lib::TestRepo;
use rstest::{fixture, rstest};
use std::path::Path;

use assert_cmd::Command;

const TRUNK: &str = "trunk";

#[fixture]
fn repo(#[default(TRUNK)] main: &str) -> TestRepo {
    let repo = TestRepo::new();
    repo.initialize(main);
    repo.commit("0.1.0-rc.1");
    repo
}

#[rstest]
fn test_default_main_branch_name(repo: TestRepo) {
    assert_version(&repo.path, &[], "0.1.0-rc.1")
}

#[rstest]
fn test_custom_main_branch(#[with("custom-main")] repo: TestRepo) {
    assert_version(&repo.path, &["--main-branch", "custom-main"], "0.1.0-rc.1")
}

#[rstest]
fn test_repository_not_in_working_directory(repo: TestRepo) {
    assert_version(".", &["--repo-path", repo.path.to_str().unwrap()], "0.1.0-rc.1")
}

fn assert_version<P: AsRef<Path>>(cd: P, args: &[&str], expected: &str) {
    let mut cmd = Command::cargo_bin(env!("CARGO_PKG_NAME"))
        .expect("CLI binary not found");

    cmd.args(args)
        .current_dir(cd)
        .assert()
        .success()
        .stdout(format!("{expected}\n"));
}