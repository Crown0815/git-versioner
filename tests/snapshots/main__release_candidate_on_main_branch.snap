---
source: tests/main.rs
description: "* 3750095 (HEAD -> trunk) 0.1.0-rc.1\n"
info:
  program: git-versioner
  args: []
---
success: true
exit_code: 0
----- stdout -----
{
  "Major": 0,
  "Minor": 1,
  "Patch": 0,
  "MajorMinorPatch": "0.1.0",
  "PreReleaseTag": "rc.1",
  "PreReleaseTagWithDash": "-rc.1",
  "PreReleaseLabel": "rc",
  "PreReleaseLabelWithDash": "-rc",
  "PreReleaseNumber": "1",
  "BuildMetadata": "",
  "SemVer": "0.1.0-rc.1",
  "AssemblySemVer": "0.1.0",
  "FullSemVer": "0.1.0-rc.1",
  "InformationalVersion": "0.1.0-rc.1"
}

----- stderr -----
